src/live_restore/live_restore_fs.c

```diff
@@ -1395,90 +1389,240 @@ __live_restore_setup_lr_fh_directory(WT_SESSION_IMPL *session, WTI_LIVE_RESTORE_
+    WT_ASSERT(session, type == WT_FS_OPEN_FILE_TYPE_LOG || type == WT_FS_OPEN_FILE_TYPE_REGULAR);
+    __wt_verbose_debug2(session, WT_VERB_LIVE_RESTORE,
+      "Transferring %s file (%s) from source to dest.\n",
+      type == WT_FS_OPEN_FILE_TYPE_LOG ? "log" : "regular", filename);
 
-    if (!dest_exist && !source_exist && !LF_ISSET(WT_FS_OPEN_CREATE))
-        WT_RET_MSG(session, ENOENT, "File %s does not exist in source or destination", name);
+    wt_off_t source_size;
+    size_t len;
+    WT_FILE_HANDLE *source_fh = NULL, *dest_fh = NULL;
 
-    if (!dest_exist && have_stop && !LF_ISSET(WT_FS_OPEN_CREATE))
-        WT_RET_MSG(session, ENOENT, "File %s has been deleted in the destination", name);
+    /* Get the full source and destination file names. */
+    WT_ERR(__live_restore_fs_backing_filename(
+      &lr_fs->source, session, lr_fs->destination.home, filename, &source_path));
+    WT_ERR(__live_restore_fs_backing_filename(
+      &lr_fs->destination, session, lr_fs->destination.home, filename, &dest_path));
 
-    /* Open it in the destination layer. */
-    WT_RET(__live_restore_fs_open_in_destination(lr_fs, session, lr_fh, name, flags, !dest_exist));
+    WT_ERR(__live_restore_remove_temporary_file(
+      session, lr_fs->os_file_system, dest_path, &tmp_dest_path));
 
-    if (have_stop || lr_fs->finished) {
-        /*
-         * Set the complete flag, we know that if there is a stop file we should never look in the
-         * source. Therefore the destination must be complete.
-         */
+    /* Open both files and create the temporary destination file. */
+    WT_ERR(lr_fs->os_file_system->fs_open_file(
+      lr_fs->os_file_system, wt_session, source_path, type, WT_FS_OPEN_EXCLUSIVE, &source_fh));
+    WT_ERR(lr_fs->os_file_system->fs_open_file(lr_fs->os_file_system, wt_session, tmp_dest_path,
+      type, WT_FS_OPEN_CREATE | WT_FS_OPEN_EXCLUSIVE, &dest_fh));
+
+    WT_ERR(
+      lr_fs->os_file_system->fs_size(lr_fs->os_file_system, wt_session, source_path, &source_size));
+
+    /*
+     * Break the copy into small chunks. Split the file into n chunks: the first n - 1 chunks will
+     * read a full read_size buffer, and the last chunk reads the remaining data.
+     */
+    WT_ERR(__wt_calloc(session, 1, read_size, &buf));
+    for (wt_off_t off = 0; off < source_size; off += (wt_off_t)len) {
+        len = WT_MIN((size_t)(source_size - off), read_size);
+        WT_ERR(source_fh->fh_read(source_fh, wt_session, off, len, buf));
+        WT_ERR(dest_fh->fh_write(dest_fh, wt_session, off, len, buf));
+
+        /* Check the system has not entered a panic state since the copy can take a long time. */
+        WT_ERR(WT_SESSION_CHECK_PANIC(wt_session));
+    }
+
+    /*
+     * Sync the file over. Then rename it so on completion it is an "atomic" operation.
+     */
+    WT_ERR(dest_fh->fh_sync(dest_fh, wt_session));
+    WT_ERR(dest_fh->close(dest_fh, wt_session));
```

Comment:
Why do we not close the source file handle?

Reply:
We about 5 lines down?

Reply:
Because the destination is the temporary file and needs closing out before we do the rename.